File: tmp/fix_jsx.js
const fs = require('fs');
const path = require('path');

const filePath = path.join('e:\\连川科技\\vetsphere', 'apps', 'admin', 'src', 'app', '(admin)', 'products', '[id]', 'page.tsx');
console.log('Path:', filePath);
console.log('Exists:', fs.existsSync(filePath));

const raw = fs.readFileSync(filePath, 'utf-8');
const lines = raw.split('\n');
console.log('Total lines:', lines.length);
console.log('Line 1038:', JSON.stringify(lines[1037]));
console.log('Line 1039:', JSON.stringify(lines[1038]));
console.log('Line 1048:', JSON.stringify(lines[1047]));

// Remove lines 1038, 1039, 1048 (0-indexed: 1037, 1038, 1047)
const toRemove = new Set([1037, 1038, 1047]);
const newLines = lines.filter((_, i) => !toRemove.has(i));

console.log('\nAfter fix:');
console.log('New total lines:', newLines.length);
for (let i = 1036; i < 1047; i++) {
  console.log(`${i+1}: ${newLines[i]}`);
}

fs.writeFileSync(filePath, newLines.join('\n'), 'utf-8');
console.log('\nFile fixed!');

File: tmp/fix_jsx.py
import sys

path = r'e:\连川科技\vetsphere\apps\admin\src\app\(admin)\products\[id]\page.tsx'

with open(path, 'rb') as f:
    raw = f.read()

print(f'File size: {len(raw)}', flush=True)
print(f'First 50 bytes: {raw[:50]}', flush=True)

has_lf = b'\n' in raw
has_crlf = b'\r\n' in raw
print(f'Has LF: {has_lf}', flush=True)
print(f'Has CRLF: {has_crlf}', flush=True)

if has_crlf:
    lines = raw.split(b'\r\n')
else:
    lines = raw.split(b'\n')

print(f'Total lines: {len(lines)}', flush=True)

if len(lines) > 1050:
    for i in range(1036, 1050):
        print(f'{i+1}: {lines[i]}', flush=True)
    
    # Remove lines: 1038 (comment), 1039 (blank), 1048 (stray })
    # 0-indexed: 1037, 1038, 1047
    to_remove = {1037, 1038, 1047}
    new_lines = [line for i, line in enumerate(lines) if i not in to_remove]
    
    sep = b'\r\n' if has_crlf else b'\n'
    with open(path, 'wb') as f:
        f.write(sep.join(new_lines))
    
    print(f'\nFixed! New total lines: {len(new_lines)}', flush=True)
    
    # Verify
    for i in range(1036, 1047):
        print(f'{i+1}: {new_lines[i]}', flush=True)
else:
    print(f'ERROR: Expected 2000+ lines but got {len(lines)}', flush=True)
